### PR TITLE
feat: images with non-root user

### PR DESCRIPTION
**Pull request by 0xgui**
In certain environments there's the need to run images as non root user. It would be nice to have that possibility and at the same time add a bit more security into the APISIX docker project.
This PR adds non root users to the docker images: alpine, centos and dashboard.

Local/Development images can still run with the root user.

### CODE REVIEW

**Review by tokers — COMMENTED** [review]
Also we use the `apisix` user as the runner of dockerfile directives, we also need to guide the users that how to configure such an user to run the APISIX processes.

File: alpine/Dockerfile [hunk 1]
```diff
@@ -17,10 +17,10 @@
 
 ARG ENABLE_PROXY=false
 
-# Build Apache APISIX
+# Build Apache APISIX image
 FROM api7/apisix-base:1.19.9.1.3
 
-ARG APISIX_VERSION=2.12.1
+ARG APISIX_VERSION=2.12.0
```
[the remark this answers is not in the document]
Comment (0xgui) [reply alpine/Dockerfile:23]: Was a mistake. Will fix it asap.

**Comment by tokers** [comment]
> 

@0xgui What's your idea about this? I didn't see any changes. 🤔

**Comment by tokers** [comment]
@0xgui So you want to change the user of APISIX's Master process to non-root, right?

**Comment by 0xgui** [comment]
> @0xgui So you want to change the user of APISIX's Master process to non-root, right?

Yes. Just like Kong does it.

**Review by tokers — APPROVED** [review]

**Pull request closed without merging**